lib.py

```diff
@@ -322,6 +332,18 @@ def get_best_times(
     return (times1, times2)
 
 
+def invalidate_submission(submission_id: SubmissionId) -> Submission:
+    """Mark a submission as invalid, and shouldn't be on the leaderboard."""
+    with Database() as db:
+        submission = db.get_submission_by_id(submission_id)
+        if submission is None:
+            logger.error("Asked to invalidate submission %s, but not found.", submission_id)
+            raise ValueError("Invalid submission.")
+        db.mark_submission_invalid(submission_id)
+
+    return submission
+
+
```

Comment:
I'm not actually certain this is worth having, but it preserves the architecture better than inlining it.